submissions/asaMitaka/FriendsApp/script.js

```diff
@@ -0,0 +1,161 @@
+const url = 'https://randomuser.me/api/?results=12';
+const content = document.querySelector('.articleBlock');
+const ageForm = document.querySelector('#ageForm');
+const genderForm = document.querySelector('#genderForm');
+const search = document.querySelector('#search');
+let friends = [];
+let friendsCopy = [];
+
+const data = (url) => fetch(url)
+    .then(handleErrors)
+    .then(res => res.json())
+    .then(res =>  {
+        friends = [...res.results];
+        init();
+    });
+
+function handleErrors(res) {
+    if (!res.ok) {
+      throw Error(res.statusText);
+    }
+    return res;
+}
+
+const filterState = {
+    search: null, 
+    gender: null,
+    sort: null,
+    reset() {
+        this.search = null;
+        this.gender = null;
+        this.sort = null;
+    }
+}
+
+function filterBy(id, usersToFilter) {
+    switch(id) {
+        case 'male':
+            usersToFilter = usersToFilter.filter(item => item.gender === 'male');
+            break;
+        case 'female':
+            usersToFilter = usersToFilter.filter(item => item.gender === 'female');
+            break;
+        case 'all':   
+            usersToFilter = [...friends];
```

Comment:
First of all, it would be more readable to create a new variable, something like `filteredUsers` in the beginning of this function to store the returned result. Secondly, if you're getting this from `friends` what's the point of passing some array in the function argument?